test/tests/functional/pbs_job_array.py

```diff
@@ -287,3 +290,276 @@ def test_qdel_expired_subjob(self):
                             "Error message is not expected")
         else:
             raise self.failureException("subjob in X state can be deleted")
+
+    def test_subjob_comments(self):
+        """
+        Test subjob comments for finished and terminated subjobs
+        """
+        a = {'resources_available.ncpus': 1}
+        self.server.manager(MGR_CMD_SET, NODE, a, self.mom.shortname)
+        j = Job(TEST_USER, attrs={
+            ATTR_J: '1-3', 'Resource_List.select': 'ncpus=1'})
+        j.set_sleep_time(5)
+        j_id = self.server.submit(j)
+        subjid_1 = j.create_subjob_id(j_id, 1)
+        subjid_2 = j.create_subjob_id(j_id, 2)
+        self.server.expect(JOB, {'comment': 'Subjob finished'}, subjid_1)
+        self.server.delete(subjid_2, extend='force')
+        self.server.expect(JOB, {'comment': 'Subjob terminated'}, subjid_2)
+        self.kill_and_restart_svr()
+        self.server.expect(JOB, {'comment': 'Subjob finished'}, subjid_1)
+
+    def test_subjob_comments_with_history(self):
+        """
+        Test subjob comments for finished, failed and terminated subjobs
+        when history is enabled
+        """
+        a = {'resources_available.ncpus': 1}
+        self.server.manager(MGR_CMD_SET, NODE, a, self.mom.shortname)
+        a = {'job_history_enable': 1}
+        self.server.manager(MGR_CMD_SET, SERVER, a)
+        j = Job(TEST_USER, attrs={
+            ATTR_J: '1-2', 'Resource_List.select': 'ncpus=1'})
+        j.set_sleep_time(5)
+        j_id = self.server.submit(j)
+        subjid_1 = j.create_subjob_id(j_id, 1)
+        subjid_2 = j.create_subjob_id(j_id, 2)
+        self.server.expect(JOB, {'comment': (
+            MATCH_RE, '.*Job run at.*and finished')}, subjid_1, extend='x')
```

Comment:
With a MATCH_RE, you don't need the initial .* (and for the other MATCH_RE initial .*)